scripts/pre-commit.sh

```diff
@@ -62,6 +64,43 @@ function run-e2e-tests() {
   ${E2E_TEST_CMD}
 }
 
+function run-e2e-upgrade-test() {
+  HOST_CLUSTER="$(kubectl config current-context)"
+
+  echo "Adding a repo to install an older kubefed version"
+  helm repo add kubefed-charts https://raw.githubusercontent.com/kubernetes-sigs/kubefed/master/charts
+  helm repo  update
+
+  # Get the previous version prior to our latest stable version
+  KUBEFED_UPGRADE_TEST_VERSION=$(helm search repo kubefed-charts/kubefed  --versions | awk '{print $2}' | head -3 | tail -1)
```

Comment:
For PRs this would currently test an upgrade from 0.5.1 to what's in the PR. However, I believe that for PRs we would like to test upgrades from the latest version (i.e. 0.6.0 atm). wdyt?

Reply:
This was testing from the old version 0.5.1 that showed the issue that this PR fixes. Generally, we should run upgrades from the most recent release, but the upgrade from 0.5.1 -> 0.6.0 is broken right now, hence this test.

I guess we should actually remove 0.6.0 once 0.6.1 is released as it's a broken release (at least for upgrades)? Thoughts?

Reply:
Yes, we should delete v0.6.0 once v0.6.1 is released.

Reply:
:+1: